src/unistd.rs

```diff
@@ -3943,11 +3943,11 @@ feature! {
     target_os = "netbsd",
     target_os = "dragonfly",
 ))]
-pub fn getpeereid(fd: RawFd) -> Result<(Uid, Gid)> {
+pub fn getpeereid<F: AsFd>(fd: &F) -> Result<(Uid, Gid)> {
```

Comment:
As discussed [here](https://github.com/nix-rust/nix/issues/1750#issuecomment-1343509652), these functions should take `F: AsFd` by value:

```rust
pub fn getpeereid<F: AsFd>(fd: F) -> Result<(Uid, Gid)> 
```

Reply:
Thanks! I just saw the discussion in #1750, will refactor now.

Reply:
https://github.com/nix-rust/nix/pull/2137/commits/22713769c030927d6a2e204430044a1497a29d95